povm_toolbox/post_processor/empirical_frequencies.py

```diff
@@ -0,0 +1,103 @@
+
+        Args:
+            loc: index of the results to use. This is relevant if multiple sets of
+                parameter values were supplied to the sampler in the same PUB. If None,
+                it is assumed that the supplied circuit was not parametrized or that a
+                unique set of parameter values was supplied. In this case, ``loc`` is
+                trivially set to 0.
+            bias: the strength of the bias towards the outcome distribution of the
+                ``ansatz`` state.  If it is a ``float``, the same bias is  applied
+                to each (local) sub-system. If it is a list of ``float``, a specific
+                bias is applied to each sub-system. If None, the bias for each sub-
+                system is set to be the number of outcomes of the POVM acting on this
+                sub-system.
+            ansatz: list of quantum states for each local sub-system, from which the
+                local outcome probability distributions are computed for each sub-
+                system. The empirical marginal frequencies are biased towards these
+                distributions. If None, the fully mixed state is used for each-subsystem.
+
+        Raises:
+            ValueError: if `loc` is None and that the POVM post-processor stores more
+                than one counter (i.e., multiple sets of parameter values were
+                supplied to the sampler in a single pub).
+            NotImplementedError: if ``self.povm`` is not a :class:`povm_toolbox.quantum_info.product_povm.ProductPOVM`
+                instance.
+        """
+        if not isinstance(self.povm, ProductPOVM):
+            raise NotImplementedError(
+                "This method is only implemented for `povm_toolbox.quantum_info.product_povm.ProductPOVM`."
+            )
+
+        if loc is None:
+            if self.counts.shape == (1,):
+                loc = (0,)
+            else:
+                raise ValueError(
+                    "`loc` has to be specified if the POVM post-processor stores"
+                    " more than one counter (i.e., if multiple sets of parameter"
+                    " values were supplied to the sampler in a single pub). The"
+                    f" array of counters is of shape {self.counts.shape}."
+                )
+
+        counts = self.counts[loc]
+        marginals = [np.zeros(subsystem_shape) for subsystem_shape in self.povm.shape]
+
+        # Computing marginals
+        shots = sum(counts.values())
+        for outcome, count in counts.items():
+            for i, k_i in enumerate(outcome):
+                marginals[i][k_i] += count / shots
+
+        alphas = []
+        # Computing alphas for each subsystem
+        for i, sub_system in enumerate(self.povm.sub_systems):
+            sub_povm = self.povm[sub_system]
+            dim = sub_povm.dimension
+            ansatz_state = DensityMatrix(np.eye(dim) / dim) if ansatz is None else ansatz[i]
```

Comment:
The length of `ansatz` should be asserted to match the length of `povm_sub_systems`